Sql storage

gatchi opened the pull request:
For use with TaskStorage

gatchi committed:
Added SQL contract

gatchi committed:
Added SQL contract

gatchi committed:
Added Database Helper

gatchi committed:
Added Database Helper

gatchi committed:
Updated db definitions

o Gave useful column names
+ Added "doneness" column
o Renamed TaskEntry and made own class
- Removed TaskSqlContract cause now empty

gatchi committed:
Updated db definitions

o Gave useful column names
+ Added "doneness" column
o Renamed TaskEntry and made own class
- Removed TaskSqlContract cause now empty

gatchi committed:
added SqlStorage, minor class edits

Version fix in TaskDbHelper.
Thorough documentation added to all classes.

gatchi committed:
added SqlStorage, minor class edits

Version fix in TaskDbHelper.
Thorough documentation added to all classes.

gatchi committed:
Javadoc edits.

Added some links and better formatting.

gatchi committed:
Javadoc edits.

Added some links and better formatting.

gatchi committed:
More formatting

gatchi committed:
More formatting

gatchi committed:
formatting

gatchi committed:
formatting

gatchi committed:
Merge branch 'sql_storage' of https://github.com/gatchimath/ToTask into sql_storage

gatchi committed:
EOF newlines

gatchi committed:
Fixing Task call syntax now that branch is rebased

Was silly and didnt rebase earlier when i should.
Ended up using an old method call (setText) that
was replaced in a separate branch and merged
into master.  This is now reflected with
setText --> setDetails in SqlStorage.java

gatchi committed:
another Task call correction

gatchi committed:
+ Adding tasks

also added exception throwing

gatchi committed:
Revert "+ Adding tasks"

This reverts commit 72e027344dcf15fd23e6bce8c2d80ad2ec2d3a7c.

mysterymath commented on app/src/main/java/com/gatchimath/totask/TaskTableEntry.java:
```diff
@@ -0,0 +1,15 @@
+ */
+public class TaskTableEntry implements BaseColumns {
+	public static final String TABLE_NAME = "tasks";
+	public static final String COLUMN_NAME_1 = "name";
```
These should be descriptive, like COLUMN_NAME_NAME, COLUMN_NAME_DETAILS, etc.

mysterymath commented on app/src/main/java/com/gatchimath/totask/TaskTableEntry.java:
```diff
@@ -0,0 +1,15 @@
+	public static final String TABLE_NAME = "tasks";
+	public static final String COLUMN_NAME_1 = "name";
+	public static final String COLUMN_NAME_2 = "details";
+	public static final String COLUMN_NAME_3 = "doneness";
```
done or done_state. "Doneness: true" makes less sense IMO.

mysterymath commented on app/src/main/java/com/gatchimath/totask/SqlStorage.java:
```diff
@@ -0,0 +1,88 @@
+ * @see com.gatchimath.totask.TaskTableEntry,
+ *      com.gatchimath.totask.TaskDbHelper
+ */
+public class SqlStorage {
```
This class is complex enough to require unit tests. You should be able to get a test context using the instrumentation library, which will allow you to set up the appropriate Context to hand to SqlStorage.

https://developer.android.com/topic/libraries/testing-support-library/index.html

mysterymath commented on app/src/main/java/com/gatchimath/totask/SqlStorage.java:
```diff
@@ -0,0 +1,88 @@
+		
+		values.put(TaskTableEntry.COLUMN_NAME_1, task.getName());
+		values.put(TaskTableEntry.COLUMN_NAME_2, task.getDetails());
+		values.put(TaskTableEntry.COLUMN_NAME_3, task.isDone() ? 1 : 0);
```
Does this work with a boolean type? If not, what's the error?

gatchi replied:
sqlite doesnt support booleans

mysterymath commented on app/src/main/java/com/gatchimath/totask/SqlStorage.java:
```diff
@@ -0,0 +1,88 @@
+	 * Returns a list of all the tasks in the table.
+	 * @return  ArrayList of Tasks in the table.
+	 */
+	public ArrayList<Task> retrieveAllTasks() {
```
readAllTasks (to parallel write)

mysterymath commented on app/src/main/java/com/gatchimath/totask/SqlStorage.java:
```diff
@@ -0,0 +1,88 @@
+		SQLiteDatabase db = dbHelper.getReadableDatabase();
+		Cursor cursor = db.query(
+			TaskTableEntry.TABLE_NAME,
+			null, null,	null, null,	null, null
```
Delete tabs between null 'groups'

mysterymath commented on app/src/main/java/com/gatchimath/totask/SqlStorage.java:
```diff
@@ -0,0 +1,88 @@
+			null, null,	null, null,	null, null
+		);
+		
+		// Now exract the info from the cursor
```
extract

mysterymath replied:
Period at the end of sentences.

mysterymath commented on app/src/main/java/com/gatchimath/totask/SqlStorage.java:
```diff
@@ -0,0 +1,88 @@
+		);
+		
+		// Now exract the info from the cursor
+		cursor.moveToFirst();
```
This may be unnecessary. You should be able to check easily once you have unit tests by deleting it and seeing if they still pass.

mysterymath commented on app/src/main/java/com/gatchimath/totask/SqlStorage.java:
```diff
@@ -0,0 +1,88 @@
+		
+		// Now exract the info from the cursor
+		cursor.moveToFirst();
+		ArrayList<Task> taskList = new ArrayList<Task>(cursor.getCount());
```
new ArrayList<>(...) is sufficient

mysterymath commented on app/src/main/java/com/gatchimath/totask/SqlStorage.java:
```diff
@@ -0,0 +1,88 @@
+		// Now exract the info from the cursor
+		cursor.moveToFirst();
+		ArrayList<Task> taskList = new ArrayList<Task>(cursor.getCount());
+		Task task;
```
Move inside while loop.

mysterymath commented on app/src/main/java/com/gatchimath/totask/SqlStorage.java:
```diff
@@ -0,0 +1,88 @@
+			task = new Task("", "");
+			
+			task.setName(cursor.getString(
+				cursor.getColumnIndexOrThrow(TaskTableEntry.COLUMN_NAME_1))
```
Get the column indices outside the loop.

mysterymath commented on app/src/main/java/com/gatchimath/totask/SqlStorage.java:
```diff
@@ -0,0 +1,88 @@
+			int taskDoneState = cursor.getInt(
+				cursor.getColumnIndexOrThrow(TaskTableEntry.COLUMN_NAME_3)
+				);	
+			if (taskDoneState == 1) task.setDoneState(true);
```
task.setDoneState(taskDoneState != 0);

gatchi committed:
formatting

gatchi committed:
Database is no longer internally generated

Must now be passed in via constructor

gatchi committed:
First JUnit test created (writeTask)

Accordingly, test sources were added to the
build.gradle.

gatchi committed:
Added Robolectric, redid SqlStorageTest

Instead of using junit tests and mockable we are
trying Robolectric.  Changes are reflected in
SqlStorageTest.java and the project build.gradle
(Robolectric does not yet support SDK version 24
yet, so a workaround was added).  The tests for
this class were also improved as well, although
theres still major work needed (retrieveAllTasks
takes 7 seconds to test somehow).

Also in this commit:
- removed sample test files (clutter)
o fixed some conditional statements flip-floppin
done status in the database of tasks

gatchi committed:
TaskStorage now tries to load list from database

TaskStorage's constructor now tries to load the
list from the database.  If it cant, the storage will
still initialize, just blank.

Also some cleanup.

gatchi committed:
Added deleteTask()

Also combined deleteTask and writeTask tests since they kinda depend on
each other, and fleshed out the retrieveAll test.

Also added a return to clearTable() (number of rows deleted)

gatchi committed:
Added SQL task deletion to TaskStorage

Also renamed addTask to add and removeTask
to remove cause thats way nicer looking and
easier to remember.

Also fixed a bug in TaskStorageTest (wasnt
initializing the storage properly).

gatchi committed:
Added generalized TaskExceptions to TaskStorage

TaskStorage and it's methods used to succeed
even if saving to another storage failed.

Now methods should fail if no other storage is
equipped (so the database erroring).
